src/scripts/inputManager.js

```diff
@@ -84,168 +89,163 @@ define(['playbackManager', 'focusManager', 'appRouter', 'dom', 'apphost'], funct
-            case 'stop':
+            },
+            'stop': () => {
                 if (checkCommandTime('stop')) {
                     playbackManager.stop();
                 }
-                break;
-            case 'changezoom':
+            },
+            'changezoom': () => {
                 playbackManager.toggleAspectRatio();
-                break;
-            case 'changeaudiotrack':
+            },
+            'changeaudiotrack': () => {
                 playbackManager.changeAudioStream();
-                break;
-            case 'changesubtitletrack':
+            },
+            'changesubtitletrack': () => {
                 playbackManager.changeSubtitleStream();
-                break;
-            case 'search':
+            },
+            'search': () => {
                 appRouter.showSearch();
-                break;
-            case 'favorites':
+            },
+            'favorites': () => {
                 appRouter.showFavorites();
-                break;
-            case 'fastforward':
+            },
+            'fastforward': () => {
                 playbackManager.fastForward();
-                break;
-            case 'rewind':
+            },
+            'rewind': () => {
                 playbackManager.rewind();
-                break;
-            case 'togglefullscreen':
+            },
+            'togglefullscreen': () => {
                 playbackManager.toggleFullscreen();
-                break;
-            case 'disabledisplaymirror':
+            },
+            'disabledisplaymirror': () => {
                 playbackManager.enableDisplayMirroring(false);
-                break;
-            case 'enabledisplaymirror':
+            },
+            'enabledisplaymirror': () => {
                 playbackManager.enableDisplayMirroring(true);
-                break;
-            case 'toggledisplaymirror':
+            },
+            'toggledisplaymirror': () => {
                 playbackManager.toggleDisplayMirroring();
-                break;
-            case 'nowplaying':
+            },
+            'nowplaying': () => {
                 appRouter.showNowPlaying();
-                break;
-            case 'repeatnone':
+            },
+            'repeatnone': () => {
                 playbackManager.setRepeatMode('RepeatNone');
-                break;
-            case 'repeatall':
+            },
+            'repeatall': () => {
                 playbackManager.setRepeatMode('RepeatAll');
-                break;
-            case 'repeatone':
+            },
+            'repeatone': () => {
                 playbackManager.setRepeatMode('RepeatOne');
-                break;
-            default:
-                break;
+            }
+        })[command];
+
+        try {
+            keyActions(commandName).call();
+        } catch (error) {
+            console.debug(`inputManager: tried to process command with no action assigned: ${commandName}`);
```

Comment:
Could it be that we have caught the exception that was thrown in the call and no one will know about it (even in browser console)?

I am not sure (don't know) about performance side of `switch` vs generating such object.

Reply:
I added that part after testing, when noticing that an "unknown" event is sent to the inputManager when focusing the window. The input manager works properly.

According to what I read, lookup tables are a lot faster on older engines and around the same nowadays. Very large switch statements (30+ items) are apparently significantly faster due to most interpreters assigning them to two arrays instead of one.

Reply:
> I added that part after testing, when noticing that an "unknown" event is sent to the inputManager when focusing the window. The input manager works properly.

I mean, if you throw an exception in `focusManager.moveUp` (or deeper) and don't catch it there. Will it be at least in console?

Reply:
I support, better to check for `keyActions(commandName)` being `undefined` explicitly and _not_ swallow the error within the handler silently